pkg/cache/cache.go

```diff
@@ -118,7 +120,19 @@ func New(defaultAudience, prefix string, clientset kubernetes.Interface) Service
 				c.addSA(sa)
 			},
 			DeleteFunc: func(obj interface{}) {
-				sa := obj.(*v1.ServiceAccount)
+				sa, ok := obj.(*v1.ServiceAccount)
+				if !ok {
+					tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+					if !ok {
+						utilruntime.HandleError(fmt.Errorf("couldn't get object from tombstone %+v", obj))
```

Comment:
Can you use `klog.Errorf` to be consistent with what we use in the rest of the code base?
If we want amazon-eks-pod-identity-webhook to use `utilruntime.HandleError` this should happen in its own PR.

Reply:
For reference, similar code in k8s/k8s.

https://github.com/kubernetes/kubernetes/blob/4ae11dac2e2cc8594c528a679db5861c0926b514/staging/src/k8s.io/sample-controller/controller.go#L357